packages/cw20/src/msg.rs

```diff
@@ -55,6 +55,8 @@ pub enum Cw20ExecuteMsg {
     /// Only with the "mintable" extension. If authorized, creates amount new tokens
     /// and adds to the recipient balance.
     Mint { recipient: String, amount: Uint128 },
+    /// Only with the "mintable" extension. The current minter may set a new minter.
+    UpdateMinter { new_minter: String },
```

Comment:
This is breaking / changing the cw20 spec. In `cw20/README.md` it says:

```
## Mintable

This allows another contract to mint new tokens, possibly with a cap.
There is only one minter specified here, if you want more complex
access management, please use a multisig or other contract as the
minter address and handle updating the ACL there.
```

So, that could be a way of doing what you need, without touching the spec.

@ethanfrey, what do you think?

Reply:
I agree with this addition.
It is not spec breaking, it is a backwards compatible addition.

we still maintain that all contracts have 0 or 1 minter, and all existing queries and messages should work the same, right?

Reply:
If there is concern, we can just add it to the cw20-base contract and not the general cw20 interface, but I don't see a big issue there.